lib/plugins/aws/invoke-local/index.js

```diff
@@ -460,7 +460,18 @@ class AwsInvokeLocal {
     // Java zip/jar implementaion doesn't preserve file permissions in a zip file.
     // https://bugs.openjdk.java.net/browse/JDK-6194856
     // So we follow Java's way in java.util.zip.ZipEntry#isDirectory() to detect directory.
-    await decompress(artifact, destination, { filter: (file) => !file.path.endsWith('/') });
+    const artifactFiles = await decompress(artifact, destination, {
+      filter: (file) => !file.path.endsWith('/'),
+    });
+    // Wrong permissions are used for rust artifacts in the following stages:
+    // 1. Artifact creation: https://github.com/softprops/serverless-rust/pull/115
+    // 2. Artifact extraction: seems like an issue in decompress package
+    // As a workaround, we are making sure the rust executable has the correct permissions
+    for (const file of artifactFiles) {
+      if (file.path === 'bootstrap') {
+        fse.chmodSync(destination + path.sep + file.path, '755');
```

Comment:
Let's not use sync method - we're using `async` versions whenever possible and resort to sync only if there's a good reason. Instead of processing in loop, let's use `.filter()` and `.map()` methods and wrap it in a `Promise.all`, it will result in something along those lines

```
await Promise.all(artifactFiles.filter(...).map(...))
```

Reply:
Instead of adding paths with `+`, let's maybe use `path.join`?

Reply:
@pgrzesik requested changes are implemented in https://github.com/serverless/serverless/pull/10891/commits/b030aac4d7f9c786fe5a883d261aa200c74a8c01. Please check it.